pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -218,7 +218,501 @@ def sipopt(instance, paramSubList, perturbList,
         kk += 1    
     
     
-    #Send the model to the ipopt_sens and collect the solution
+    # Send the model to the ipopt_sens and collect the solution
     results = opt.solve(m, keepfiles=keepfiles, tee=streamSoln)
 
     return m
+
+
+
+def kaug(instance, paramSubList, perturbList,
+         cloneModel=True, streamSoln=False, keepfiles=False, optarg=None):
+    """This function modifies def ipopt() to use kaug. All documents are taken 
+    from documents in def sipopt(). This function accepts a Pyomo 
+    ConcreteModel, a list of parameters, along with their corresponding 
+    perturbation list. The model is then converted into the design 
+    structure required to call kaug dsdp mode to get an approximate 
+    perturbed solution with updated bounds on the decision variable. 
+    
+    Parameters
+    ----------
+    instance: ConcreteModel
+        pyomo model object
+
+    paramSubList: list
+        list of mutable parameters
+
+    perturbList: list
+        list of perturbed parameter values
+
+    cloneModel: bool, optional
+        indicator to clone the model. If set to False, the original
+        model will be altered
+
+    streamSoln: bool, optional
+        indicator to stream IPOPT solution
+
+    keepfiles: bool, optional
+        preserve solver interface files
+            
+    optarg : dictionary, optional
+        solver options dictionary object (default=None)
+    
+    Returns
+    -------
+    model: ConcreteModel
+        The model modified for use with kaug.  
+        The model contains the approximated results at the perturbation point
+        
+    Raises
+    ------
+    ValueError
+        perturbList argument is expecting a List of Params
+    ValueError
+        length(paramSubList) must equal length(perturbList)
+    ValueError
+        paramSubList will not map to perturbList
+    ImportError
+        ipopt binary must be available
+    ImportError
+        k_aug binary must be available
+    ImportError
+        dotsens binary must be available
+    Exception
+        kaug does not support inequality constraints
+    """
+
+    # Verify User Inputs    
+    if len(paramSubList)!=len(perturbList):
```

Comment:
Is this code repeated/duplicated in the sipopt function? If so, you should either merge these functions or move these checks to a stand-alone function that is called by `sipopt` and `kaug`.